### PR TITLE
Ensure that Az.psm1 is signed and used by Az.psd1 as RootModule

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Fix for https://github.com/Azure/azure-powershell/issues/7563

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/232/

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/233/

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/234/

File: .gitignore [hunk 1]
```diff
@@ -242,4 +242,3 @@ launchSettings.json
 /SolutionMappings.json
 /TestMappings.json
 /tools/Modules/tmp
-/tools/Az/Az.psm1
```
Comment (MiYanni) [inline .gitignore]: Don't remove it from .gitignore. See https://github.com/Azure/azure-powershell/issues/7603

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/243/

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign/247/

**Pull request merged**